### ISSUE TITLE
Adding weights is "undirected", getting weights "directed"

### ISSUE THREAD

**Issue by karajan9** [sbromberger/MetaGraphs.jl#60]
When I change a weight
```
using LightGraphs
using MetaGraphs
g = MetaGraph(2)
add_edge!(g, 1, 2)
set_prop!(g, 2, 1, :weight, 7.0)
```
the edge is treated as "undirected":
```
set_props!(g::MetaGraph, _e::SimpleEdge, d::Dict)
    e = LightGraphs.is_ordered(_e) ? _e : reverse(_e)
    [...]
```
If I then try to retrieve the value via ```get_prop``` vs  ```weights```
```
get_prop(g, 1, 2, :weight) == 7.0
get_prop(g, 2, 1, :weight) == 7.0
weights(g)[1, 2] == 7.0
weights(g)[2, 1] == 1.0 (defaultweight)
```
```weights``` seems to treat the edges as "directed", which results in
```
dijkstra_shortest_paths(g, 1).dists[2] == 7.0
dijkstra_shortest_paths(g, 2).dists[1] == 1.0 (defaultweight)
```
without the graph being intentionally directed.
 
I use Julia v0.7 with LightGraphs v1.2.0 and MetaGraphs v0.6.0. If there are any questions, I'm happy to help.

**Comment by sbromberger** [comment]
definitely a bug, and should be an easy fix.

**Comment by karajan9** [comment]
I would try my luck fixing it (never done a PR before) if you could tell me what the intended behavior is -- I have basically no clue about the internals of LightGraphs or MetaGraphs and the consequences on other parts if I try to fix this. Should ```set_props``` treat the edges as directed or should ```weights``` treat them as undirected or should the behavior depend on some property (however, I wouldn't know which)?

**Comment by sbromberger** [comment]
This would make a great first PR, and I can help you with it (though there may be a delay due to holidays).

`weights` should treat the edges as undirected for undirected graphs. You can use the behavior of `SimpleWeightedGraphs` as a comparison.

**Comment by karajan9** [comment]
> This would make a great first PR, and I can help you with it (though there may be a delay due to holidays).

Sure, delays shouldn't be a problem, the same goes for me.

> ```weights``` should treat the edges as undirected for undirected graphs. You can use the behavior of ```SimpleWeightedGraphs``` as a comparison.

Alright, thanks for the hint and I will get back to you when I have something working or could use some help. :smiley:

**Comment by karajan9** [comment]
I created a PR for a some tests for this issue, mostly to test my setup I created with some help of the friendly people over at Slack.

During that I encountered
```
mdg60 = MetaDiGraph(2)
add_edge!(mdg60, 1, 2)
get_prop(mdg60, 1, 2, :weight)
```
throws an error. Is that correct or should it return the default weight?

Lastly, I have two possible solutions for the original issue. My plan was to create PRs for both and then discuss them with you?

**Comment by sbromberger** [comment]
that should throw an error. The correct way to get a weight is `weight()`. This is merely querying a dict for an arbitrary key.

**Comment by karajan9** [comment]
Alright.

Well, I didn't know that PR automatically update when I push to the branch so that PR became now the first suggestion. -> #61 

The other suggestion is over at #62

**Comment by sbromberger** [comment]
Closed via #62. Thank you!
